### PR TITLE
feat: support release commits (and improve tests)

### LINKED ISSUE #19: Support for setting commit
Release tracking gets even better with Commit Hash info when the GitHub source repo is integrated with Sentry.

The `set-commits` command would be useful if integrated

### PR DESCRIPTION
**Pull request by vaind**
## :scroll: Description
- refactor: speed up plugin tests by using mock CLI
- test: refactor plugin_test
- feat: support setting commits - closes #19 
- fix integration test server (wip)

## :green_heart: How did you test it?

added unit tests and manually on a personal project

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 * Add support to load release variable from environment ([#40](https://github.com/getsentry/sentry-dart-plugin/pull/40))
 * Download Sentry CLI on first run ([#49](https://github.com/getsentry/sentry-dart-plugin/pull/49))
+* Support release commits ([#62](https://github.com/getsentry/sentry-dart-plugin/pull/62))
```
Comment (marandaneto) [inline CHANGELOG.md:9]: Move to unreleased section

File: lib/src/configuration.dart [hunk 2]
```diff
@@ -58,6 +57,8 @@ class Configuration {
   /// the Web Build folder, defaults to build/web
   late String webBuildFilesFolder;
 
+  late String commits;
```
Comment (marandaneto) [inline lib/src/configuration.dart:60]: Add a /// comment

File: test/plugin_test.dart [hunk 3]
```diff
@@ -14,50 +16,119 @@ void main() {
...
-version: 1.1.0
+name: $project
+version: $version
 
 sentry:
-  upload_native_symbols: true
-  include_native_sources: true
-  upload_source_maps: true
-  auth_token: t
+  auth_token: t # TODO: support not specifying this, let sentry-cli use the value it can find in its configs
```
Comment (marandaneto) [inline test/plugin_test.dart:46]: The problem was that this plugin could not log a proper error, sentry CLI outputs too much stuff when something goes wrong.

File: test/plugin_test.dart [hunk 4]
```diff
@@ -14,50 +16,119 @@ void main() {
...
+  }
+
+  test('fails without args and pubspec', () async {
+    final exitCode = await plugin.run([]);
+    expect(exitCode, 1);
+    expect(pm.commandLog, const ['chmod +x $cli', '$cli help']);
+  });
+
+  test('works with pubspec', () async {
+    // TODO: because `url` param affects all commands, make it a test-group argument and run all test cases with/without it.
```
Comment (marandaneto) [inline test/plugin_test.dart:65]: Something that we should address in this PR?

**Comment by marandaneto** [comment]
@vaind Amend the README, https://github.com/getsentry/sentry-dart-plugin/blob/main/README.md#configuration-optional
So users know how to use it.

File: test/plugin_test.dart (hunk 3 above)
Comment (vaind) [reply test/plugin_test.dart:46]: That can be solved too. I just didn't want to do this unrelated thing in the same PR.

Also relates to https://github.com/getsentry/sentry-dart-plugin/issues/23

Comment (marandaneto) [reply test/plugin_test.dart:46]: Yes, but I didn't want to during the hack week, we can leave this open and fix it later.
My message was more about giving context on why it's like this.

File: test/plugin_test.dart (hunk 4 above)
Comment (vaind) [reply test/plugin_test.dart]: didn't want to blow this up too much but since you've already seen tests without that change, I've done it now.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**